pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/PinotDataType.java

```diff
@@ -355,6 +368,15 @@ public String toString(Object value) {
   }
 
   public Integer[] toIntegerArray(Object value) {
+    if (value instanceof List) {
```

Comment:
You need to use the singleValueType to do the type conversion. Please refer to the code from line 383-390, and use the same naming convention